### PR TITLE
feat(helm): Detailed exit code for helm plugins

### LINKED ISSUE #4170: Allow returning specific exit code for helm plugins
We are looking into returning different exit codes in the helm-diff to simplify some CI scenarios (See databus23/helm-diff#54)

The issue is that Helm considers any non zero exit codes from the plugin to be an error and will always exit with `1`. We need to be able to return a specific exit code to the caller.

E.g. we want the `helm diff upgrade` command to return with exit code `2` if no error occurred but some changes were found.

Any thoughts on how this could/should be implemented?

### PR DESCRIPTION
**Pull request by mumoshu**
Resolves #4170

### CODE REVIEW

File: cmd/helm/helm.go [hunk 1]
```diff
@@ -158,7 +158,12 @@ func init() {
 func main() {
 	cmd := newRootCmd(os.Args[1:])
 	if err := cmd.Execute(); err != nil {
-		os.Exit(1)
+		switch e := err.(type) {
```
Comment (mumoshu) [inline cmd/helm/helm.go]: This should be an idiomatic way for this kind of switch as explained in https://github.com/pkg/errors/issues/73#issuecomment-244641060

File: cmd/helm/load_plugins.go [hunk 2]
```diff
@@ -87,7 +93,11 @@ func loadPlugins(baseCmd *cobra.Command, out io.Writer) {
 				if err := prog.Run(); err != nil {
 					if eerr, ok := err.(*exec.ExitError); ok {
 						os.Stderr.Write(eerr.Stderr)
-						return fmt.Errorf("plugin %q exited with error", md.Name)
+						status := eerr.Sys().(syscall.WaitStatus)
```
Comment (mumoshu) [inline cmd/helm/load_plugins.go]: Even though this sounds like *unix, it [works on windows too](https://stackoverflow.com/a/10385867).

**Review by technosophos — APPROVED** [review]
LGTM

**Comment by technosophos** [comment]
Trying to trick the CLA bot into re-running the check by adding the string "I signed it" into the comment body.

**Pull request merged**